idea/src/org/jetbrains/kotlin/idea/intentions/branchedTransformations/IfThenUtils.kt

```diff
@@ -223,6 +225,24 @@ data class IfThenToSelectData(
     internal fun getImplicitReceiver(): ImplicitReceiver? = baseClause.getResolvedCall(context)?.getImplicitReceiverValue()
 
     internal fun hasImplicitReceiver(): Boolean = getImplicitReceiver() != null
+
+    private fun replacedBaseClause(baseClause: KtCallExpression, receiver: KtExpression, factory: KtPsiFactory): KtExpression {
+        return factory.buildExpression {
+            appendExpression(receiver)
+            appendFixedText("?.let {")
+            appendExpression(baseClause.calleeExpression)
+            appendFixedText("(")
+            baseClause.valueArguments.forEachIndexed { index, arg ->
+                if (index != 0) appendFixedText(", ")
+                val argExpression = arg.getArgumentExpression()
+                if (argExpression?.evaluatesTo(receiverExpression) == true)
+                    appendFixedText("it")
```

Comment:
What if we have already `it` in name scope. Is it possible to get a name conflict and/or semantic change?